src/pages/view-only.tsx

```diff
@@ -0,0 +1,144 @@
+import { Box, ChakraProvider } from '@chakra-ui/react';
+import { useInterpret, useMachine } from '@xstate/react';
+import Head from 'next/head';
+import { NextRouter, useRouter } from 'next/router';
+import React, { useEffect, useMemo } from 'react';
+import { createMachine, MachineConfig, sendParent } from 'xstate';
+import { CanvasProvider } from '../CanvasContext';
+import { CanvasView } from '../CanvasView';
+import { EmbedProvider } from '../embedContext';
+import { SimulationProvider } from '../SimulationContext';
+import { simModel, simulationMachine } from '../simulationMachine';
+import { theme } from '../theme';
+import { NextComponentWithMeta } from '../types';
+import { useInterpretCanvas } from '../useInterpretCanvas';
+import { parseEmbedQuery, withoutEmbedQueryParams } from '../utils';
+import { withReadyRouter } from '../withReadyRouter';
+import lzString from 'lz-string';
+
+const parseMachineFromQuery = (query: NextRouter['query']) => {
+  if (!query.machine) {
+    throw new Error();
+  }
+
+  if (Array.isArray(query.machine)) {
+    throw new Error();
+  }
+
+  const lzResult = lzString.decompressFromEncodedURIComponent(query.machine);
+
+  if (!lzResult) throw new Error();
+
+  const machineConfig = JSON.parse(lzResult);
+
+  // Tests that the machine is valid
+  return createMachine(machineConfig);
```

Comment:
optionally you could extend the message with `err.message`
```suggestion
  try {
    return createMachine(machineConfig);
  } catch {
    throw new Error("decompressed `machine` couldn't be used to `createMachine`")
  }
```